interpreter/src/functions.rs

```diff
@@ -223,6 +224,17 @@ pub fn starts_with(This(this): This<Arc<String>>, prefix: Arc<String>) -> bool {
     this.starts_with(prefix.as_str())
 }
 
+/// Returns true if a string matches the regular expression.
+///
+/// # Example
+/// ```cel
+/// "abc".matches("^[a-z]*$") == true
+/// ```
+pub fn matches(This(this): This<Arc<String>>, regex: Arc<String>) -> bool {
+    let re = Regex::new(&regex).unwrap();
```

Comment:
Let's make sure to use the built-in error handling.

You will need to accept `FunctionContext` as a parameter
https://github.com/clarkmcc/cel-rust/blob/45c883d5fc4c8a26f200ce8bbf3d8ae75ab2624a/interpreter/src/functions.rs#L148

Then you will need to use the function context to build an error which will have more information about what function is being called, etc.
https://github.com/clarkmcc/cel-rust/blob/45c883d5fc4c8a26f200ce8bbf3d8ae75ab2624a/interpreter/src/functions.rs#L157

Reply:
Are you asking why `starts_with` doesn't take `FunctionContext` as it's first parameter? `starts_with` isn't fallible so it doesn't need to return an error.

Reply:
There might be some helpful examples here: [functions.rs](https://github.com/clarkmcc/cel-rust/blob/master/example/src/functions.rs)

Reply:
Thanks, yeah... I played a bit with the different error conditions and I think I get what behavior we want.